test/end-to-end/cases/label-element.xspec

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<x:description query="x-urn:test:do-nothing" query-at="../../do-nothing.xquery"
+	schematron="../../do-nothing.sch" stylesheet="../../do-nothing.xsl"
+	xmlns:x="http://www.jenitennison.com/xslt/xspec">
+
+	<!--
+		xspec-utils_stylesheet.xspec uses this file for testing x:is-user-content().
+		When modifying this file, check whether xspec-utils_stylesheet.xspec needs any additions or updates.
+	-->
+
+	<!--
+		* '&#x09;&#x0A;&#x0D;&#x20;' is inserted deliberately to see what will happen.
+		  * In particular, //x:label/text()[position() = (1, 3, last())] are deliberately made as
+		    whitespace-only text nodes.
+		
+		* Comments and processing instructions inserted deliberately to verify these:
+		  * XSpec schema allows them.
+		  * XSpec compiler ignores them.
+	-->
+
+	<x:scenario>
+		<x:label>&#x09;&#x0A;&#x0D;&#x20;<!--test-->&#x09;&#x0A;&#x0D;&#x20;My<?test test?>&#x09;&#x0A;&#x0D;&#x20;<!--test-->Scenario&#x09;&#x0A;&#x0D;&#x20;<?test test?>&#x09;&#x0A;&#x0D;&#x20;</x:label>
+		<x:call function="true" />
+		<x:expect test="false()">
+			<x:label>&#x09;&#x0A;&#x0D;&#x20;<!--test-->&#x09;&#x0A;&#x0D;&#x20;My<?test test?>&#x09;&#x0A;&#x0D;&#x20;<!--test-->Expectation&#x09;&#x0A;&#x0D;&#x20;<?test test?>&#x09;&#x0A;&#x0D;&#x20;</x:label>
+		</x:expect>
+	</x:scenario>
+
+	<x:scenario label="Test with x:like/@label">
+		<x:like
+			label="&#x09;&#x0A;&#x0D;&#x20;&#x09;&#x0A;&#x0D;&#x20;My&#x09;&#x0A;&#x0D;&#x20;Scenario&#x09;&#x0A;&#x0D;&#x20;&#x09;&#x0A;&#x0D;&#x20;"
+		 />
+	</x:scenario>
+
+	<x:scenario label="Test with x:like/x:label">
+		<x:like>
+			<!-- This x:label is exactly the same as /x:description/x:scenario[1]/x:label -->
```

Comment:
Because it's exactly the same, this file doesn't really demonstrate that the XSpec compiler ignores comments and processing instructions when matching `x:like/x:label` to `x:scenario/x:label`. If that's part of what you were intending to demonstrate, consider making a comment and PI on line 38 a little different from line 22.

Reply:
Thanks. I wasn't aware of that point. Comments and PIs changed in 5aa75fc9b5d981910c4a6bf6c68c0b3a413b3cfc.

Reply:
Thanks, I like your latest change. :)